packages/dev/core/src/Rendering/geometryBufferRenderer.ts

```diff
@@ -360,14 +364,102 @@ export class GeometryBufferRenderer {
+                        defines.push("#define SPECULARGLOSSINESSTEXTURE");
+                        defines.push(`#define REFLECTIVITY_UV${material.specularGlossinessTexture.coordinatesIndex + 1}`);
+                        needUv = true;
+                    } else {
+                        if (material.specularColor != null) {
+                            defines.push("#define REFLECTIVITYCOLOR");
+                        }
+                    }
+                    if (material.glossiness != null) {
+                        defines.push("#define GLOSSINESSS");
+                    }
+                } else if (material instanceof PBRMaterial) {
+                    // if it is the bigger PBRMaterial
+                    if (material.metallicTexture != null) {
+                        defines.push("#define ORMTEXTURE");
+                        defines.push(`#define REFLECTIVITY_UV${material.metallicTexture.coordinatesIndex + 1}`);
+                        defines.push("#define METALLICWORKFLOW");
+                        needUv = true;
+                        metallicWorkflow = true;
+                    }
+                    if (material.metallic != null) {
+                        defines.push("#define METALLIC");
+                        defines.push("#define METALLICWORKFLOW");
+                        metallicWorkflow = true;
+                    }
+
+                    if (material.roughness != null) {
+                        defines.push("#define ROUGHNESS");
+                        defines.push("#define METALLICWORKFLOW");
+                        metallicWorkflow = true;
+                    }
+
+                    if (metallicWorkflow) {
+                        if (material.albedoTexture != null) {
+                            defines.push("#define ALBEDOTEXTURE");
+                            defines.push(`#define ALBEDO_UV${material.albedoTexture.coordinatesIndex + 1}`);
+                            needUv = true;
+                        } else if (material.albedoColor != null) {
+                            defines.push("#define ALBEDOCOLOR");
+                        }
+                    } else {
+                        // SpecularGlossiness Model
+                        if (material.reflectivityTexture != null) {
+                            defines.push("#define SPECULARGLOSSINESSTEXTURE");
+                            defines.push(`#define REFLECTIVITY_UV${material.reflectivityTexture.coordinatesIndex + 1}`);
+                            needUv = true;
+                        } else if (material.reflectivityColor != null) {
+                            defines.push("#define REFLECTIVITYCOLOR");
+                        }
+                        if (material.microSurface != null) {
+                            defines.push("#define GLOSSINESSS");
+                        }
+                    }
+                } else if (material instanceof StandardMaterial) {

```

Comment:
Replace instanceof by getClassName() for speed purpose

Reply:
Thank you very much for all your feedback!